StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetError.swift

```diff
@@ -34,14 +66,74 @@ extension PaymentSheetError: CustomDebugStringConvertible {
+        case .paymentIntentInTerminalState(status: let status):
+            return "PaymentSheet received a PaymentIntent in a terminal state: \(status)"
+        case .setupIntentInTerminalState(status: let status):
+            return "PaymentSheet received a SetupIntent in a terminal state: \(status)"
+        case .fetchPaymentMethodsFailure:
+            return "Failed to retrieve PaymentMethods for the customer"
+        case .linkSignUpNotRequired:
+            return "Don't call sign up if not needed"
+        case .noPaymentMethodTypesAvailable(intentPaymentMethods: let intentPaymentMethods):
+            return "None of the payment methods on the PaymentIntent/SetupIntent can be used in PaymentSheet: \(intentPaymentMethods). You may need to set `allowsDelayedPaymentMethods` or `allowsPaymentMethodsRequiringShippingAddress` in your PaymentSheet.Configuration object."
+        case .linkCallVerifyNotRequired:
+            return "Don't call verify if not needed"
+        case .linkingWithoutValidSession:
+            return "Linking account session without valid consumer session"
+        case .savingWithoutValidLinkSession:
+            return "Saving to Link without valid session"
+        case .payingWithoutValidLinkSession:
+            return "Paying with Link without valid session"
+        case .deletingWithoutValidLinkSession:
+            return "Deleting Link payment details without valid session"
+        case .updatingWithoutValidLinkSession:
+            return "Updating Link payment details without valid session"
+        case .linkLookupNotFound:
+            return "Link account not found"
+        case .failedToCreateLinkSession:
+            return "Failed to create Link account session"
+        case .linkNotAuthorized:
+            return "confirm called without authorizing Link"
+        case .setupIntentClientSecretProviderNil:
+            return "setupIntentClientSecretForCustomerAttach, but setupIntentClientSecretProvider is nil"
+        case .unknown:
+            return "unknown"
+        }
+    }
+
+    /// A description logged to a developer for debugging
     public var debugDescription: String {
-        return "An error occured in PaymentSheet. " + {
-            switch self {
-            case .noPaymentMethodTypesAvailable(let intentPaymentMethods):
-                return "None of the payment methods on the PaymentIntent/SetupIntent can be used in PaymentSheet: \(intentPaymentMethods). You may need to set `allowsDelayedPaymentMethods` or `allowsPaymentMethodsRequiringShippingAddress` in your PaymentSheet.Configuration object."
-            case .unknown(let debugDescription):
-                return debugDescription
-            }
-        }()
+        switch self {
+        case .unknown(debugDescription: let debugDescription):
+            return "An error occured in PaymentSheet. " + debugDescription
+        default:
+            return "An error occured in PaymentSheet. " + safeLoggingString
```

Comment:
for case .unknown, maybe it would be better to have:
```
case .unknown:
return "An unknown error occurred in PaymentSheet. "
default:
return "An error occurred in PaymentSheet. "
```

also note that `occurred` has two `r` 's

Reply:
Updated, looks like occurred was misspelled for quite a while here lol